pkg/tms/tms.go

```diff
@@ -412,3 +473,94 @@ func upload(communicationInstance *CommunicationInstance, uploadRequestData pipe
 	defer response.Body.Close()
 	return data, nil
 }
+
+func UnmarshalServiceKey(serviceKeyJson string) (serviceKey serviceKey, err error) {
+	err = json.Unmarshal([]byte(serviceKeyJson), &serviceKey)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func JsonToMap(jsonStr string) map[string]interface{} {
+	result := make(map[string]interface{})
+	json.Unmarshal([]byte(jsonStr), &result)
+	return result
+}
+
+func FormNodeIdExtDescriptorMappingWithValidation(utils TmsUtils, nodeNameExtDescriptorMapping map[string]interface{}, nodes []Node, mtaYamlMap map[string]interface{}, mtaVersion string) (map[int64]string, error) {
```

Comment:
There is no test for this?

Reply:
already commented above

Reply:
> There is no test for this?

FormNodeIdExtDescriptorMappingWithValidation covered by almost all tests in tmsUpload_test.go and tmsExport_test.go